core/src/test/java/org/elasticsearch/action/admin/cluster/node/tasks/TasksIT.java

```diff
@@ -429,7 +429,7 @@ public void testTasksListWaitForTimeout() throws Exception {
                 .execute();
         try {
             // Wait for the task to start on all nodes
-            assertBusy(() -> assertEquals(internalCluster().numDataAndMasterNodes(), client().admin().cluster().prepareListTasks()
+            assertBusy(() -> assertEquals(internalCluster().size(), client().admin().cluster().prepareListTasks()
```

Comment:
this change would have been ok even without your PR right? The difference between size and numDataAndMasterNodes is the coordinating only nodes which we are not touching here?